Gvi to points

ioalexei opened the pull request:
Addresses issue #8 

Adds `file assign_gvi_to_points.py`

Input: 
1. mapillary images directory
2. interim points file generated by `create_points.py`
3. an output directory (I've been using `data/processed` as default)

Output:
The module will output a GeoPackage file called `image_points_with_gvi.gpkg` in the output directory. 

Example usage: 
```
python3 -m src.assign_gvi_to_points /data/raw/mapillary /data/interim/sample_images.gpkg /data/processed
```

Still to do - flesh out the info/examples for `--help`

danbjoseph commented:
I was able to run things successfully. 
- Please also update `pyproject.toml` with the new dependencies.
- In previous steps, instead of just the output directory, it also takes the filename and I think it will write in different geospatial file format according to the extension. Since the process might be run more than once or on multiple datasets, it would be good to be able to set the filename in the command.

Something seems off (either with the methodology or with the code) based on a spot check of the 3 highest and 3 lowest GVI scores.

## High
81.8
![781053043909949](https://github.com/AmericanRedCross/street-view-green-view/assets/4806884/df77ff4b-5e44-4b53-b8a7-5884882c89fc)
81  
![713345330995536](https://github.com/AmericanRedCross/street-view-green-view/assets/4806884/aa42a641-08b1-4362-97da-58d871fd94d0)
80.8
![425963426515579](https://github.com/AmericanRedCross/street-view-green-view/assets/4806884/c8e806d5-37be-4cad-b9e0-96f02b7ceb5c)


## Low
2
![422966303445031](https://github.com/AmericanRedCross/street-view-green-view/assets/4806884/3c2d429e-1ddd-466f-9ef5-2e3397a24666)

1.2
![246682028517324](https://github.com/AmericanRedCross/street-view-green-view/assets/4806884/72cd685c-19c0-426a-82cb-2daa99bd2f3e)

1.14
![935156744512812](https://github.com/AmericanRedCross/street-view-green-view/assets/4806884/5525c290-efb8-4f75-80ee-b6dda5eeda1d)

ioalexei commented:
Updated: 
* New dependencies added to pyproject.toml
* Formatting applied
* Changed output directory to output file - user can enter a file path and user-defined filename/format for the output

ioalexei commented:
Spot checks definitely look odd. I didn't touch the GV score function but will take a look and compare to the sample from Treepedia to see if I can spot anything off.

dragonejt commented on src/assign_gvi_to_points.py:
```diff
@@ -0,0 +1,107 @@
+
+@app.command()
+def main(image_directory, interim_data, output_file):
+    """
```
nit: You can use `Annotated` to build out the help message. See https://github.com/AmericanRedCross/street-view-green-view/blob/main/src/download_images.py#L132.

ioalexei replied:
Thanks - added help annotations

dragonejt reviewed (changes requested):
Please address the `os` built-in library being in `pyproject.toml`, the rest of my comments are nits that you can choose to implement or not. Thanks so much for your work!

dragonejt commented on pyproject.toml:
```diff
@@ -26,7 +26,10 @@ dependencies = [
   "tqdm",
-  "python-dotenv"
+  "python-dotenv", 
+  "os", 
```
Our "Install Dependencies" step in the Code Quality workflow is failing because of this: https://github.com/AmericanRedCross/street-view-green-view/actions/runs/8442740425/job/23222294773?pr=29#step:4:50

ioalexei commented:
Thanks @dragonejt - should all be addressed now.
